src/recipes/FactoryBuilding.ts

```diff
@@ -69,6 +73,25 @@ export const FactoryConveyorBelts = sortBy(
   'name',
 );
 
+export const FactoryPipelines = sortBy(
+  // Includes clean variant as well tbd.
+  AllFactoryBuildings.filter(
+    building =>
+      building.pipeline?.isPipeline && building.pipeline?.flowRate > 0,
+  ), // Only return buildable which actually have a flow rate
+  'name',
+);
+
+export const FactoryPipelinesExclAlternates = sortBy(
+  AllFactoryBuildings.filter(
+    building =>
+      building.pipeline?.isPipeline &&
+      building.pipeline?.flowRate > 0 &&
```

Comment:
`&& building.pipeline!.flowRate` (we are sure that pipeline is present, and without `!` ts will warn about a potential null vs number comparison)